geps/gep-2643/index.md

```diff
@@ -0,0 +1,514 @@
+server name is identified, the gateway does a full encrypted passthrough of the 
+communication, this GEP will also cover cases where a TLS communication is 
+terminated on the Gateway before being passed to a backend.
+
+## Goals
+
+* Provide a TLS route type, based on the SNI identification.
+* Support both Passthrough (core support) and Termination (extended support) TLS modes for TLSRoute
+* Provide load balancing of a TLS route, allowing a user to define a route, 
+based on the SNI identification that should pass the traffic to N load balanced backends
+
+## Longer Term Goals
+* Implement capabilities for [ESNI](https://www.cloudflare.com/learning/ssl/what-is-encrypted-sni/)
+* Implement capabilities for [ECH](https://blog.cloudflare.com/encrypted-client-hello/)
+* Extend TLSRouteRule to support hostname being a matcher
+
+## Non-Goals
+* Provide an interface for users to define different listeners or ports for the
+`TLSRoute` - This will be covered by the [ListenerSet enhancement](https://gateway-api.sigs.k8s.io/geps/gep-1713/).
+* Describe re-encryption capabilities when TLS Termination is being used - This will
+be covered by [GEP-4274](https://github.com/kubernetes-sigs/gateway-api/issues/4274), extending
+`BackendTLSPolicy` to allow `TLSRoute`.
+* When using `TLSRoute` passthrough, support `PROXY` protocol on the gateway listener.
+* When using `TLSRoute` passthrough, support `PROXY` protocol on the communication
+between the Gateway and the backend.
+* Support TLS over UDP or UDP-based protocols such as `QUIC`.
+* Support attributes other than the SNI/hostname for the route selection
+* Support multiplexing (HTTPS and TLS protocols) on the same Listener - This should be covered
+on [GEP-4271](https://github.com/kubernetes-sigs/gateway-api/issues/4271)
+
+## Introduction
+
+While many application routing cases can be implemented using HTTP/L7 matching
+(the tuple `protocol:hostname:port:path`), there are some specific cases where direct,
+encrypted communication to the backend may be required, without further assertion. For example:
+* A backend that is TLS based but not HTTP based (e.g., a Kafka service, or a
+Postgres service, with its listener being TLS enabled).
+* Some WebRTC solutions.
+* Backends that can require direct client-certificate authentication (e.g., OAuth).
+
+For the example cases above, it is desired that the routing is made as a passthrough
+mode, where the `Gateway` passes the packets to the backend without terminating TLS.
+
+On some other cases, it is desired that the termination is done on the `Gateway`
+and the proxy passes the unencrypted packets to the backend without caring about 
+attributes other than a TCP communication.
+
+While this routing is possible using other mechanisms such as a `TCPRoute` or a
+Kubernetes service of type `LoadBalancer`, it may be desired to have a single point
+of traffic convergence (like one single `LoadBalancer`) for reasons like cost,
+traffic control, etc.
```

Comment:
One single LB is a great idea, but can you explain why this is only possible using SNI and TLSRoute?

Reply:
the point here is not about being possible just using SNI and TLSRoute. It is about being DESIRED to have a single point of traffic convergence for non HTTPs traffic.

eg.: 
- I have 3 services that are Postgres. They all listen on port 5432. They all are TLS enabled. I can create a service of type LoadBalancer for each of them, but this will incur in costs
- I have 3 services that are Postgres. Same situation as above. Instead, I could use TCPRoute, but then I would need different ports for each service on my listener (because TCPRoute is classified only by the dstip:dstport)
- Same scenario as above. I can use the SNI attribute (postgres1.tld, postgres2.tld, postgres3.tld) to listen the traffic all on the same port (5432) but classify it based on the SNI attribute